### PR TITLE
Blog App forms

### PR DESCRIPTION
**Pull request by Isabellanamutenda**
In this pull request I:

> Create forms to perform these functions:

- Create a post on behalf of the current_user
- Create a comment on behalf of the current_user

> Allow users to add likes to posts

This is my last commit I hope I have done all the changes required please consider

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,6 @@ yarn-debug.log*
 
 # Ignore folder
 /vendor
+
+
+
```
Comment (DeliceLydia) [inline .gitignore:42]: I could find any changed codes here but the project requires to:
- [ ] Add a comment to a post but currently am getting an error
d

- [ ] A user should be able to add a like to posts. please work on `likes` controller and it should have a create actions

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Isabellanamutenda, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,13 +1,10 @@
 class UsersController < ApplicationController
   def index
-    if current_user
-      @users = User.all
-    else
-      redirect_to user_session_path
-    end
+    @users = User.all
```
Comment (dasileker) [inline app/controllers/users_controller.rb:3]: - [x] please verify the user can log in. now can't

kkkk

**Review by dasileker — CHANGES_REQUESTED** [review]
## **Hi @Isabellanamutenda,** 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (Isabellanamutenda) [reply app/controllers/users_controller.rb:3]: Fixed

File: app/models/user.rb [hunk 3]
```diff
@@ -1,29 +1,9 @@
 class User < ApplicationRecord
```
Comment (SajjadAhmad14) [inline app/models/user.rb:1]: User is not able to sign in. Please make sure a user is able to sign in.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -1,39 +1,28 @@
 class PostsController < ApplicationController
```
Comment (SajjadAhmad14) [inline app/controllers/posts_controller.rb:1]: Please add a route in the config/routes.rb file for creating like. User should be able to like posts.

File: app/models/user.rb [hunk 5]
```diff
@@ -1,29 +1,9 @@
...
-  include Devise::JWT::RevocationStrategies::JTIMatcher
```
Comment (SajjadAhmad14) [inline app/models/user.rb:2]: User is unable to `seed` users data. Kindly see the below error.
seed

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -1,13 +1,10 @@
...
   end
 
   def show
```
Comment (SajjadAhmad14) [inline app/controllers/users_controller.rb:6]: Please see the error below which occurs when user clicks on a user profile.
users_show

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -1,39 +1,28 @@
...
+    @posts = @user.posts
   end
 
   def show
-    @user = User.find(params[:user_id])
-    @post = @user.posts.includes(:comments, :likes).find(params[:id])
+    @post = Post.find(params[:id])
   end
 
   def new
```
Comment (SajjadAhmad14) [inline app/controllers/posts_controller.rb:12]: User is not able to create a post. Kindly make sure that a signed in user can create a post.

File: app/controllers/comments_controller.rb [hunk 8]
```diff
@@ -1,12 +1,12 @@
 class CommentsController < ApplicationController
   skip_before_action :verify_authenticity_token
-  load_and_authorize_resource
   def new
```
Comment (SajjadAhmad14) [inline app/controllers/comments_controller.rb:3]: User is not able to comment on a post. Please check that a user is able to comment on a post.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Ashley, 
 ### You have been doing a tremendous job so far :ok_hand:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/comments_controller.rb (hunk 8 above)
Comment (Isabellanamutenda) [reply app/controllers/comments_controller.rb:3]: Fixed

File: app/controllers/posts_controller.rb (hunk 7 above)
Comment (Isabellanamutenda) [reply app/controllers/posts_controller.rb:12]: Fixed

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (Isabellanamutenda) [reply app/controllers/users_controller.rb:6]: Fixed

**Pull request closed without merging**